### ISSUE TITLE
[BUG] StorageBlobCreatedEventData in NuGet package seems to have int ContentLength, instead of documented long

### ISSUE THREAD

**Issue by TMPhoenix** [Azure/azure-sdk-for-net#15856]
**Describe the bug**
While deserializing an event grid message (on a storage queue) about a large uploaded blob, we got an overflow exception. Investigating, I found that the reason was the StorageBlobCreatedEventData class had an int ContentLength, which differs from the long defined here in the code and in the documentation, thus limiting the max size of Blobs to about 2GB...

Temporarily worked around by defining a class with the same signature, but with the expected "long?" ContentLength.

**Expected behavior**
NuGet package defines StorageBlobCreatedEventData with correct "long?" ContentLength, allowing for normal deserialization of message.

**Actual behavior (include Exception or Stack Trace)**
OverflowException while NewtonSoft Json tries to place a number larger than maxint into an integer ContentLength.

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

1. Setup event grid to report created event to a storage queue for a blob container
2. Upload a file larger then ~3GB
3. Deserialize the message and get the event data from it. Try to convert the event data to StorageBlobCreatedEventData.

**Environment:**
 - Name and version of the Library package used: Microsoft.Azure.EventGrid 3.2.0
 - Hosting platform or OS and .NET runtime version: Linux docker image running on Azure AppService, .NET Core 3.1
 - IDE and version : Visual Studio 2019

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team best able to assist.

**Comment by jfggdl** [comment]
@TMPhoenix . Thanks for reporting this issue. We will be trying to reproduce the problem and will report our findings.

**Comment by jfggdl** [comment]
@TMPhoenix, would you please provide the stack trace (sequence of calls that leads to the error)? Thanks

**Comment by TMPhoenix** [comment]
Here you go:
<details>
<summary>Click to Expand</summary>
[
    {
        "outerId": "0",
        "message": "Exception while executing function: xxx_ACTIVITY",
        "parsedStack": [
            {
                "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
                "level": 0,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<ExecuteWithLoggingAsync>d__18.MoveNext",
                "level": 1,
                "fileName": "C:\\projects\\azure-webjobs-sdk-rqm4t\\src\\Microsoft.Azure.WebJobs.Host\\Executors\\FunctionExecutor.cs",
                "assembly": "Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35",
                "line": 303
            },
            {
                "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
                "level": 2,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification",
                "level": 3,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<TryExecuteAsync>d__15.MoveNext",
                "level": 4,
                "fileName": "C:\\projects\\azure-webjobs-sdk-rqm4t\\src\\Microsoft.Azure.WebJobs.Host\\Executors\\FunctionExecutor.cs",
                "assembly": "Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35",
                "line": 140
            }
        ],
        "type": "Microsoft.Azure.WebJobs.Host.FunctionInvocationException",
        "severityLevel": "Error",
        "id": "1814293"
    },
    {
        "outerId": "1814293",
        "message": "Could not convert to integer: 2777262080. Path 'contentLength'.",
        "parsedStack": [
            {
                "method": "Newtonsoft.Json.JsonReader.ReadAsInt32",
                "level": 0,
                "assembly": "Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed",
                "line": 0
            },
            {
                "method": "Newtonsoft.Json.JsonReader.ReadForType",
                "level": 1,
                "assembly": "Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed",
                "line": 0
            },
            {
                "method": "Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject",
                "level": 2,
                "assembly": "Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed",
                "line": 0
            },
            {
                "method": "Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject",
                "level": 3,
                "assembly": "Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed",
                "line": 0
            },
            {
                "method": "Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal",
                "level": 4,
                "assembly": "Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed",
                "line": 0
            },
            {
                "method": "Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize",
                "level": 5,
                "assembly": "Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed",
                "line": 0
            },
            {
                "method": "Newtonsoft.Json.JsonSerializer.DeserializeInternal",
                "level": 6,
                "assembly": "Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed",
                "line": 0
            },
            {
                "method": "Newtonsoft.Json.Linq.JToken.ToObject",
                "level": 7,
                "assembly": "Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed",
                "line": 0
            },
            {
                "method": "Newtonsoft.Json.Linq.JToken.ToObject",
                "level": 8,
                "assembly": "Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed",
                "line": 0
            },
            {
                "method": "Newtonsoft.Json.Linq.JToken.ToObject",
                "level": 9,
                "assembly": "Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed",
                "line": 0
            },
            {
                "method": "xxxxxxxxx.Platform.BackendFunctions.Functions.ComputeOrchestration.ComputeResultConsumer+<ProcessMessageBatchAsync>d__6.MoveNext",
                "level": 10,
                "fileName": "C:\\Users\\xxx\\source\\repos\\Platform\\src\\xxxxxxxxx.Platform.BackendFunctions\\Functions\\ComputeOrchestration\\ComputeResultConsumer.cs",
                "assembly": "xxxxxxxxx.Platform.BackendFunctions, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null",
                "line": 118
            },
            {
                "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
                "level": 11,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification",
                "level": 12,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "xxxxxxxxx.Platform.BackendFunctions.Functions.Eternals.EternalRunner`2+<InvokeAsync>d__18.MoveNext",
                "level": 13,
                "fileName": "C:\\Users\\xxx\\source\\repos\\Platform\\src\\xxxxxxxxx.Platform.BackendFunctions\\Functions\\Eternals\\EternalRunner.cs",
                "assembly": "xxxxxxxxx.Platform.BackendFunctions, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null",
                "line": 67
            },
            {
                "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
                "level": 14,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification",
                "level": 15,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult",
                "level": 16,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "xxxxxxxxx.Platform.BackendFunctions.Functions.ComputeOrchestration.ComputeResultConsumer+<RunActivity>d__13.MoveNext",
                "level": 17,
                "fileName": "C:\\Users\\xxx\\source\\repos\\Platform\\src\\xxxxxxxxx.Platform.BackendFunctions\\Functions\\ComputeOrchestration\\ComputeResultConsumer.cs",
                "assembly": "xxxxxxxxx.Platform.BackendFunctions, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null",
                "line": 195
            },
            {
                "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
                "level": 18,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification",
                "level": 19,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "Microsoft.Azure.WebJobs.Host.Executors.FunctionInvoker`2+<InvokeAsync>d__10.MoveNext",
                "level": 20,
                "fileName": "C:\\projects\\azure-webjobs-sdk-rqm4t\\src\\Microsoft.Azure.WebJobs.Host\\Executors\\FunctionInvoker.cs",
                "assembly": "Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35",
                "line": 53
            },
            {
                "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
                "level": 21,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification",
                "level": 22,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<InvokeAsync>d__26.MoveNext",
                "level": 23,
                "fileName": "C:\\projects\\azure-webjobs-sdk-rqm4t\\src\\Microsoft.Azure.WebJobs.Host\\Executors\\FunctionExecutor.cs",
                "assembly": "Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35",
                "line": 542
            },
            {
                "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
                "level": 24,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification",
                "level": 25,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<ExecuteWithWatchersAsync>d__25.MoveNext",
                "level": 26,
                "fileName": "C:\\projects\\azure-webjobs-sdk-rqm4t\\src\\Microsoft.Azure.WebJobs.Host\\Executors\\FunctionExecutor.cs",
                "assembly": "Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35",
                "line": 508
            },
            {
                "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
                "level": 27,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification",
                "level": 28,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<ExecuteWithLoggingAsync>d__24.MoveNext",
                "level": 29,
                "fileName": "C:\\projects\\azure-webjobs-sdk-rqm4t\\src\\Microsoft.Azure.WebJobs.Host\\Executors\\FunctionExecutor.cs",
                "assembly": "Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35",
                "line": 446
            },
            {
                "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
                "level": 30,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification",
                "level": 31,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<ExecuteWithLoggingAsync>d__18.MoveNext",
                "level": 32,
                "fileName": "C:\\projects\\azure-webjobs-sdk-rqm4t\\src\\Microsoft.Azure.WebJobs.Host\\Executors\\FunctionExecutor.cs",
                "assembly": "Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35",
                "line": 303
            }
        ],
        "type": "Newtonsoft.Json.JsonReaderException",
        "severityLevel": "Error",
        "id": "36385898"
    },
    {
        "outerId": "36385898",
        "message": "Value was either too large or too small for an Int32.",
        "parsedStack": [
            {
                "method": "System.Convert.ThrowInt32OverflowException",
                "level": 0,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "System.Convert.ToInt32",
                "level": 1,
                "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
                "line": 0
            },
            {
                "method": "Newtonsoft.Json.JsonReader.ReadAsInt32",
                "level": 2,
                "assembly": "Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed",
                "line": 0
            }
        ],
        "type": "System.OverflowException",
        "severityLevel": "Error",
        "id": "10025342"
    }
]
</details>

**Comment by jfggdl** [comment]
@TMPhoenix . Thank you. We are already working on identifying the root cause of this. Stay tuned.

**Comment by ahamad-MS** [comment]
Hello @TMPhoenix ... Thanks for reaching out and reporting this issue. 

Is this issue still repro'ing for you? From (https://github.com/Azure/azure-sdk-for-net/blob/2b594335737a65cb8f36543ef817deaba5c43f4b/sdk/eventgrid/Azure.Messaging.EventGrid/src/Generated/Models/StorageBlobCreatedEventData.cs#L58 ) it seems the content length is long and not int anymore.

Please confirm.

@kerri-lee  Can you please help here or help find appropriate owner for updating this file if needed.

**Comment by kerri-lee** [comment]
Hi @ahamad-MS, I think @TMPhoenix is using the Microsoft.Azure.EventGrid library. The content length also seems to be long and not int (https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventgrid/Microsoft.Azure.EventGrid/src/Generated/Models/StorageBlobCreatedEventData.cs#L125), but I'm not exactly sure how the deserialization works for that library.

@JoshLove-msft could you help me with this?

**Comment by JoshLove-msft** [comment]
Hi @kerri-lee! Yes that's right - this issue is about the Microsoft.Azure.EventGrid library and it is not an issue in the new [Azure.Messaging.EventGrid](https://www.nuget.org/packages/Azure.Messaging.EventGrid/4.0.0-beta.4) library that is in beta.

**Comment by ahamad-MS** [comment]
Yes @kerri-lee .. you are right. I meant the Microsoft.Azure.EventGrid library.

@TMPhoenix  Please confirm if the issue still repro for you. thanks

**Comment by TMPhoenix** [comment]
@ahamad-MS - I'm not at the office the coming week, so I can't easily check, however, once I get back I can have a look if it is still happening. Just to be clear we worked around the problem by replicating the message class with the corrected signature and using that to deserialize the json message that event grid has placed in a storage queue (or rather the part of the message that the StorageBlobCreatedEventData represents). However it would feel better if we could avoid using a custom class to do so.

**Comment by TMPhoenix** [comment]
I have taken another look at this. At first I thought the problem was still there, but since I couldn't quite figure out why it was happening I decided to go through all our projects and add the latest version of the Microsoft.Azure.EventGrid nuget package as an explicit dependency  (the project that does the deserialization already had the latest version as an explicit dependency). That seems to have cleared it up. Even though we were building our docker image with the latest version (I checked the version in the docker image), for some reason some projects that should be unrelated to the code at hand were pulling in an ancient version (1.1.3 I believe) during compilation.

I still don't quite understand why the older version was somehow being used during compilation despite the latest version being the explicit dependency, but the issue at hand seems to have been resolved.
